### PR TITLE
Use eslint and fix linter issues

### PR DESCRIPTION
**Pull request by mantoni**
As noted in #5, I'd like to use eslint going forward. I've copied the rules from another project I maintain where the code style is similar enough to not have too many linter issues. With this change, the build should pass again.

### CODE REVIEW

**Comment by mantoni** [comment]
Ok, the timezone tests fail on travis. They work for me locally, because I'm in "the right" timezone. Once this got merged, PR #2 should fix the tests.

**Comment by mantoni** [comment]
I merged #2 and rebased here. Now there's a new issue with the tests. I think this should be merged and the tests fixed in a new PR.

**Comment by maks** [comment]
Yes sorry about that, I think the test fails are due to a typo I made in the timestamp formatting fix, I'll try to fix this.

File: test/item-processor-test.js [hunk 1]
```diff
@@ -43,9 +43,10 @@ test('processor', {
     assert.equal(this.item.dates.fullDate.created, 'January 1st 1970');
     assert.equal(this.item.dates.fullDate.modified, 'January 2nd 1970');
     assert.equal(this.item.dates.fullDate.rendered, 'January 3rd 1970');
-    assert.equal(this.item.dates.someTime.created, timezoneOffSet+':00:00');
-    assert.equal(this.item.dates.someTime.modified, (timezoneOffSet+1)+':00:00');
-    assert.equal(this.item.dates.someTime.rendered,  (timezoneOffSet+2)+':00:00');
+    // zero-pad, ref: https://groups.google.com/forum/#!topic/nodejs/df5UzXFAByA
+    assert.equal(this.item.dates.someTime.created, ("0" + timezoneOffSet).slice(-2)+':00:00');
+    assert.equal(this.item.dates.someTime.modified, ("0" + (timezoneOffSet+1)).slice(-2) +':00:00');
+    assert.equal(this.item.dates.someTime.rendered, ("0" + (timezoneOffSet+2)).slice(-2)+':00:00');
```
Comment (mantoni) [inline test/item-processor-test.js]: Well, this will solve it if the timezone is has 0 to 9 hours offset ...

**Comment by mantoni** [comment]
Eslint has some stylistic complaints 😄  (see failing build).

**Comment by maks** [comment]
Yes sorry just fixed them now

**Review by mantoni — COMMENTED** [review]
👍

File: test/item-processor-test.js (hunk 1 above)
Comment (maks) [reply test/item-processor-test.js]: actually it works for upto 2 digits, which is enough as luckily for now we only need to worry about earth timezones :smile:

Comment (mantoni) [reply test/item-processor-test.js]: Haha. True. I just saw this as well and merged. Unfortunately, the build fails on my box now with 1 hour offset (CEST) ...

Comment (maks) [reply test/item-processor-test.js]: Drats, sorry! I finally looked up how to test with any TZ so will do a new PR to fix this.

**Pull request merged**